build/chip/write_build_time_header.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python
+# Copyright (c) 2023 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import optparse
```

Comment:
https://docs.python.org/3/library/optparse.html

OptParse is deprecated since 3.2. Any chance to use argparse?

Reply:
yup. Definitely copied this nearly directly from the other script, which appears to be older. Same reasoning for below. Both changed.